### ISSUE TITLE
sometimes sample returns `None` 

### ISSUE THREAD

**Issue by koaning** [jmschrei/pomegranate#162]
So I'm building a markov chain to come up with new IKEA furniture names (because reasons). Turns out that `pomegrenate` might be a nice helper for the job. 

```
from pomegranate import *
import itertools as it

poke_names = file("/Users/code/Development/grabbel/demo_data/ikea_furniture.txt").read().split('\n')

def flatten(listOfLists):
    "Flatten one level of nesting"
    return it.chain.from_iterable(listOfLists)

tokens = flatten([ [t for t in name.replace(' ', '').replace("'", '')] for name in poke_names])
uniq_tokens = list(set(tokens))

product2 = [list(i) + [1.0/len(uniq_tokens)**2] for i in it.product(uniq_tokens, repeat=2)]
product3 = [list(i) + [1.0/len(uniq_tokens)**3] for i in it.product(uniq_tokens, repeat=3)]

d1 = DiscreteDistribution({t:1.0/len(uniq_tokens) for t in uniq_tokens})
d2 = ConditionalProbabilityTable(product2, [d1])
d3 = ConditionalProbabilityTable(product3, [d1, d2])

mc = MarkovChain(3)
mc.fit(poke_names, inertia = 0.01)

for i in range(20): 
    print("".join(mc.sample(5)))
```

When I run this though, I sometimes get an error suggesting that `mc.sample` sometimes generates `None` values. How is this possible? I've added the Ikea furniture data. 

[ikea_furniture.txt](https://github.com/jmschrei/pomegranate/files/459735/ikea_furniture.txt)

**Comment by mikegraham** [comment]
I'm confused about the line `mc = MarkovChain(3)` -- that's how how `MarkovChain` works, right?

**Comment by jmschrei** [comment]
No, it's not currently how it works. Currently you pass in a list of distributions of increasing parent size until you reach the maximum number of parents a distribution can have. I think he was suggesting an alternate initialization if you wanted to learn the chain entirely from data by specifying only the number of positions back to use, as opposed to pass in pre-defined distributions.

**Comment by mikegraham** [comment]
Thanks for following up. I think I  must have misinterpreted the intent of the snippet -- I thought it was supposed to be a reproducing case, not a suggestion.

**Comment by jmschrei** [comment]
Actually I'm a little confused now too. @koaning can you clarify? As @mikegraham noticed, `mc = MarkovChain(3)` is not how you initialize that. Is this what you're doing in your code? It should be `mc = MarkovChain([d1, d2, d3])`. Doing what you did should raise an error, which is a separate issue I should address.

**Comment by jmschrei** [comment]
I just added a method to allow you to write `model = MarkovChain.from_samples(X, weights=None, k=3)` to learn a model directly from data without the need to add in intermediary dummy tables.
